cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowStatsSlaHttpHandlerTest.java

```diff
@@ -0,0 +1,140 @@
+import org.apache.http.HttpResponse;
+import org.apache.twill.api.RunId;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Tests for {@link co.cask.cdap.gateway.handlers.WorkflowStatsSlaHttpHandler}
+ */
+public class WorkflowStatsSlaHttpHandlerTest extends AppFabricTestBase {
+
+  private static final Gson GSON = new Gson();
+  public static final Supplier<File> TEMP_FOLDER_SUPPLIER = new Supplier<File>() {
+
+    @Override
+    public File get() {
+      try {
+        return tmpFolder.newFolder();
+      } catch (IOException e) {
+        throw Throwables.propagate(e);
+      }
+    }
+  };
+
+  @Test
+  public void testStatistics() throws Exception {
+
+    Store store = getInjector().getInstance(DefaultStore.class);
+    final ApplicationWithPrograms app = AppFabricTestHelper.deployApplicationWithManager(WorkflowApp.class,
+                                                                                         TEMP_FOLDER_SUPPLIER);
+    Iterable<Program> programs = app.getPrograms();
+    String workflowName = null;
+    String mapreduceName = null;
+    String sparkName = null;
+
+    for (Program program : programs) {
+      if (program.getType() == ProgramType.WORKFLOW) {
+        workflowName = program.getName();
+      } else if (program.getType() == ProgramType.MAPREDUCE) {
+        mapreduceName = program.getName();
+      } else if (program.getType() == ProgramType.SPARK) {
+        sparkName = program.getName();
+      }
+    }
+
+    Assert.assertNotNull(workflowName);
+    Assert.assertNotNull(mapreduceName);
+    Assert.assertNotNull(sparkName);
+
+    Id.Program workflowProgram =
+      Id.Workflow.from(Id.Namespace.DEFAULT, "WorkflowApp", ProgramType.WORKFLOW, workflowName);
+    Id.Program mapreduceProgram =
+      Id.Program.from(Id.Namespace.DEFAULT, "WorkflowApp", ProgramType.MAPREDUCE, mapreduceName);
+    Id.Program sparkProgram =
+      Id.Program.from(Id.Namespace.DEFAULT, "WorkflowApp", ProgramType.SPARK, sparkName);
+
+    for (long i = 0; i < 100; i++) {
+      RunId workflowRunId = RunIds.generate();
+      store.setStart(workflowProgram, workflowRunId.getId(), RunIds.getTime(workflowRunId, TimeUnit.SECONDS));
+
+      long randomValue = (long) (Math.random() * 50);
+      Thread.sleep(randomValue);
+
+      RunId mapreduceRunid = RunIds.generate();
+      store.setWorkflowProgramStart(mapreduceProgram, mapreduceRunid.getId(), workflowProgram.getId(),
+                                    workflowRunId.getId(), mapreduceProgram.getId(),
+                                    RunIds.getTime(mapreduceRunid, TimeUnit.SECONDS), null, null);
+      store.setStop(mapreduceProgram, mapreduceRunid.getId(),
+                    RunIds.getTime(workflowRunId, TimeUnit.SECONDS) + randomValue, ProgramRunStatus.COMPLETED);
+      Thread.sleep(randomValue);
+
+      if ((Math.random() * 3) == 0) {
```

Comment:
is this the intended behavior? if((int)(Math.random() * 3) == 0) or if(Math.random() < 0.3333333)